test-data/boids-module.ron

```diff
@@ -0,0 +1,2325 @@
+                class: Input,
+                binding: Some(BuiltIn(GlobalInvocationId)),
+                ty: (
+                    index: 9,
+                    marker: (),
+                ),
+            ),
+        ],
+    ),
+    functions: (
+        data: [
+            (
+                name: Some("vtx_main"),
+                control: (
+                    bits: 0,
+                ),
+                parameter_types: [],
+                return_type: None,
+                global_usage: [
+                    (
+                        bits: 1,
+                    ),
+                    (
+                        bits: 1,
+                    ),
+                    (
+                        bits: 1,
+                    ),
+                    (
+                        bits: 2,
+                    ),
+                ],
+                local_variables: (
+                    data: [
+                        (
+                            name: Some("angle"),
+                            ty: (
+                                index: 3,
+                                marker: (),
+                            ),
+                            init: None,
+                        ),
+                        (
+                            name: Some("pos"),
+                            ty: (
+                                index: 1,
+                                marker: (),
+                            ),
+                            init: None,
+                        ),
+                    ],
+                ),
+                expressions: (
+                    data: [
+                        GlobalVariable((
+                            index: 3,
+                            marker: (),
+                        )),
+                        GlobalVariable((
+                            index: 4,
+                            marker: (),
+                        )),
+                        GlobalVariable((
+                            index: 1,
+                            marker: (),
+                        )),
+                        GlobalVariable((
+                            index: 2,
+                            marker: (),
+                        )),
+                        AccessIndex(
+                            base: (
+                                index: 4,
+                                marker: (),
+                            ),
+                            index: 0,
+                        ),
+                        AccessIndex(
+                            base: (
+                                index: 4,
+                                marker: (),
+                            ),
+                            index: 1,
+                        ),
+                        Call(
+                            name: "atan2",
+                            arguments: [
+                                (
+                                    index: 5,
```

Comment:
It looks to me that most of the space in this RON is dedicated to `Handle<>` serialization.
Would you be open to augment it a bit in the same way `wgpu` deals with Ids?
See https://github.com/gfx-rs/wgpu/blob/0a0ef9a1000a4d327209570dd2606008b133a2e3/wgpu-core/src/id.rs#L22-L29

In this case, it would need to look like `Id(index)`

Reply:
I gave this a shot but I had trouble getting `serde_derive` to work with the `PhantomData`. Do you think it should be possible with `serde_derive`? Or possibly I could try implementing `Serialize` and `Deserialize` by hand for `Handle<T>`?

Reply:
The code I linked to does almost precisely this, it also has `PhantomData`. Implementing the traits manually is also OK to me, but I think the same approach as linked is slightly easier.

Reply:
Yeah, you're right. I'll take another look to see what I am getting wrong.

Reply:
All right, I implemented and tested the concise handle serialization, then pushed it onto this PR. Fun trick 😀